pkg/component/component.go

```diff
@@ -452,6 +452,15 @@ func PushLocal(client *occlient.Client, componentName string, applicationName st
 	}
 	targetPath := fmt.Sprintf("%s/src", s2iSrcPath)
 
+	stdin := bytes.NewBuffer([]byte{})
+	stdout := bytes.NewBuffer([]byte{})
+	stderr := bytes.NewBuffer([]byte{})
+	err = client.ExecCMDInContainer(pod.Name, strings.Split("/var/lib/supervisord/bin/supervisord ctl start run", " "), stdout, stderr, stdin, false)
```

Comment:
If possible we  should split `s2i-setup` and `run` from `setup-and-run`.
Setup could be run only when the container is started, and the Supervisor can just execute `run` script without setup.